src/relay/op/tensor/transform.cc

```diff
@@ -1637,28 +1637,30 @@ Array<Tensor> LayoutTransformCompute(const Attrs& attrs,
 
   CHECK(src_layout.defined() && dst_layout.defined())
     << "cannot convert from/to undefined layout";
-  CHECK(src_layout.Convertible(dst_layout))
+
+  auto layout_converter = BijectiveLayoutNode::make(src_layout, dst_layout);
+  CHECK(layout_converter.defined())
     << "cannot convert from " << param->src_layout << " to " << param->dst_layout;
 
-  const auto& out_shape = ConvertLayout(inputs[0]->shape, src_layout, dst_layout);
+  const auto& out_shape = layout_converter.ForwardShape(inputs[0]->shape);
   return Array<Tensor> {
       topi::layout_transform(inputs[0], out_shape, [&](const Array<tvm::Var>& dst_indices) {
```

Comment:
Can we implement this logic in topi?

Reply:
ah, yes, forgot to modify. it can be implemented by `ForwardIndex` actually.